lib/utils.js

```diff
@@ -114,3 +114,20 @@ export function when (items) {
 
   return allResolvedPromise
 }
+
+export function PausableTimer (callback, delay) {
```

Comment:
I like this function but since it's being used with `new` it could be a class in ES6.

```
export class PausableTimer {
  constructor (callback, delay) {
    ...
  }

  pause () {
    ...
  }

  resume () {
    ...
  }
}
```